openedx/core/djangoapps/programs/utils.py

```diff
@@ -226,12 +264,38 @@ def completed_course_runs(self):
         Returns:
             list of dicts, each representing a course run certificate
         """
+        return self.course_runs_with_state['completed_runs']
```

Comment:
Doubling up cached properties seems weird to me. You don't need this method or `failed_course_runs` now that you have a property that returns both values. All you're doing at this point is adding method calls.